text_generation/causal_lm/cpp/continuous_batching/library/src/logit_processor.hpp

```diff
@@ -23,6 +23,10 @@ class ILogitTransformer {
 public:
     virtual std::vector<Token> apply(const std::vector<Token>& input_logits) = 0;
 
+    bool is_applicable(size_t generated_tokens_cnt) {
+        return generated_tokens_cnt < m_applicable_tensor_len;
```

Comment:
`ILogitTransformer` is a generic class, while this logic with `m_applicable_tensor_len ` EOS transformer specific.

In future we plan to add an ability to expose this interface to end users, so it should be more clear:
- Can `apply` change its input to avoid copy each time? It can be very important even now, because vocab size is typically at least 50k and >100k for newer models.
- In this case, EOS transformer can just do nothing and check its applicability condition within its body